### ISSUE TITLE
SqlitePropertyBuilderExtensions have wrong generic constraints on PropertyBuilder<T> extensions

### ISSUE THREAD

**Issue by rowanmiller** [aspnet/EntityFramework#4469]
The correct generic is `PropertyBuilder<TProperty>` (i.e. no constraints on the type of T)

```
public static PropertyBuilder<TProperty> ForSqlServerHasDefaultValueSql<TProperty>(this PropertyBuilder<TProperty> propertyBuilder, string sql) 
public static PropertyBuilder<TProperty> HasDefaultValueSql<TProperty>(this PropertyBuilder<TProperty> propertyBuilder, string sql) 
```

However SQLite extensions are defined as `PropertyBuilder<TEntity>` and TEntity is constrained to a reference type. The T should be renamed and the constraint removed.

```
public static PropertyBuilder<TEntity> ForSqliteHasDefaultValueSql<TEntity>(this PropertyBuilder<TEntity> builder, string sql) where TEntity : class;
```

This causes compilation errors if you try to set a default value for SQLite on a property that is a value type.

**Comment by rowanmiller** [comment]
BTW applies to all extensions, not just `HasDefaultValueSql`

**Comment by smitpatel** [comment]
Should I update this in my current PR for default value?

**Comment by rowanmiller** [comment]
Sure :smile:

**Comment by smitpatel** [comment]
Fixed in 2948069534c9a99aecba796f3a4697c5ca7d75a9
